codegens/nodejs-superagent/lib/parseRequest.js

```diff
@@ -0,0 +1,165 @@
+            '${sanitize(filePath, trimBody)}')\n`;
+        });
+        if (fileSnippet !== '') {
+          accumulator.push(fileSnippet);
+        }
+        else {
+          return accumulator;
+        }
+      }
+      else if (typeof item.src !== 'string') {
+        accumulator.push(indentString + `.attach('${sanitize(item.key, trimBody)}', '/path/to/file')`);
+      }
+      else {
+        var pathArray = item.src.split(path.sep),
+          fileName = pathArray[pathArray.length - 1];
+        accumulator.push(indentString + `.attach('${sanitize(item.key, trimBody)}', 
+              '${sanitize(item.src, trimBody)}', '${sanitize(fileName, trimBody)}')`);
+      }
+    }
+    else if (requestType === 'urlencoded') {
+      accumulator.push(indentString +
+        `.send({'${sanitize(item.key, trimBody)}':'${sanitize(item.value, trimBody)}'})`);
+    }
+    else if (requestType === 'formdata') {
+      accumulator.push(indentString +
+        `.field('${sanitize(item.key, trimBody)}', '${sanitize(item.value, trimBody)}')`);
+    }
+    return accumulator;
+  }, []);
+  return snippetString.join('\n');
+}
+
+/**
+ * Parses body object based on mode of body and returns code snippet
+ *
+ * @param {Object} requestBody - json object for body of request
+ * @param {String} indentString - string for indentation
+ * @param {Boolean} trimBody - indicates whether to trim body fields or not
+ * @param {String} contentType Content type of the body being sent
+ */
+function parseBody (requestBody, indentString, trimBody, contentType) {
+  if (requestBody) {
+    var bodySnippet = indentString;
+    switch (requestBody.mode) {
+      case 'raw':
+        if (contentType === 'application/json') {
+          try {
+            let jsonBody = JSON.parse(requestBody[requestBody.mode]);
+            bodySnippet += `.send(JSON.stringify(${JSON.stringify(jsonBody)}))\n`;
+            break;
+          }
+          catch (error) {
+            bodySnippet += `.send(${JSON.stringify(requestBody[requestBody.mode])})\n`;
+            break;
+          }
+        }
+        bodySnippet += `.send(${JSON.stringify(requestBody[requestBody.mode])})\n`;
+        break;
+      // eslint-disable-next-line no-case-declarations
+      case 'graphql':
+        let query = requestBody[requestBody.mode].query,
+          graphqlVariables;
+        try {
+          graphqlVariables = JSON.parse(requestBody[requestBody.mode].variables);
+        }
+        catch (e) {
+          graphqlVariables = {};
+        }
+        bodySnippet += '.send(JSON.stringify({\n' +
+          `${indentString.repeat(2)}query: '${sanitize(query, trimBody)}',\n` +
+          `${indentString.repeat(2)}variables: ${JSON.stringify(graphqlVariables)}\n` +
+          `${indentString}}))\n`;
+        break;
+      case 'formdata':
+        bodySnippet = `${extractFormData(requestBody[requestBody.mode], indentString, trimBody, requestBody.mode)}`;
```

Comment:
a `\n` is missing at the end for formdata case

Reply:
Thanks for pointing this out @shreys7
I have added a commit to correct this ✔️

Reply:
Tests are failing because of #246